src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -869,6 +912,12 @@ ImmutableArray<CompilationEvent> dequeueGeneratedCompilationEvents()
 
                 while (_compilation.EventQueue.TryDequeue(out CompilationEvent compilationEvent))
                 {
+                    if (compilationEvent is CompilationStartedEvent compilationStartedEvent &&
+                        _analysisOptions.Options?.AdditionalFiles.Length > 0)
+                    {
+                        compilationEvent = compilationStartedEvent.WithAdditionalFiles(_analysisOptions.Options.AdditionalFiles);
```

Comment:
>compilationEvent = compilationStartedEvent.WithAdditionalFiles(_analysisOptions.Options.AdditionalFiles); [](start = 24, length = 105)

Why didn't we enqueue the event with `AdditionalFiles`? Does this mean others callers of `TryDequeue()` are also required to fix up `CompilationStartedEvent`?

Reply:
> Why didn't we enqueue the event with AdditionalFiles

Enqueue is done within the core compiler layer, while AdditionalFiles are only available in the analyzer driver/CompilationWithAnalyzers layer. We can potentially tag along `(compilationStartedEvent, additionalFiles)` here, but that was making the API signatures too messy. This is the only place where CompilationWithAnalyzers dequeues and processes a compilationStartedEvent, so it shouldn't be required elsewhere.